extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -1071,6 +1065,38 @@ export const popupMessageListener = (request: Request, sessionStore: Store) => {
     return { error: "Session timed out" };
   };
 
+  const addToken = async () => {
+    const privateKey = privateKeySelector(sessionStore.getState());
+    const networkDetails = await getNetworkDetails();
+
+    const Sdk = getSdk(networkDetails.networkPassphrase);
+
+    if (privateKey.length) {
+      const sourceKeys = Sdk.Keypair.fromSecret(privateKey);
+      const tokenInfo = tokenQueue.pop();
+
+      if (!tokenInfo?.contractId) {
+        throw Error("Missing contract id");
+      }
+
+      const response = await addTokenWithContractId({
+        contractId: tokenInfo.contractId,
+        network: networkDetails.network,
+        publicKey: sourceKeys.publicKey(),
```

Comment:
an easier way to get the public key is to just use `publicKeySelector`. That way, you don't need to bother with the private key at all. In fact, it's preferable because I'm about to open a PR that removes `privateKeySelector` 😃

Reply:
https://github.com/stellar/freighter/pull/1815/commits/2188240c6033a6ba5d3409c14dc3f86899345f30